### PR TITLE
Add support for eth/69 (EIP-7642)

### LINKED ISSUE #10659: Add support for eth/69 (EIP-7642)
### Describe the feature

ref https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7642.md

## TODO
* add types for eth/69 first, then check what we'd need to check on p2p impl

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by malik672**
this still like the half incomplete part of the code
@mattsse I have ideas on how they should be implemented but I don't think they might necessarily be what is wanted

Closes #10659

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
this is hard to review because it contains unrelated changes (your previous pr)

this pr should only focus on new types and not change any implementation logic because 69 is not used yet

File: crates/net/eth-wire-types/src/version.rs [hunk 1]
```diff
@@ -24,20 +24,23 @@ pub enum EthVersion {
...
     Eth68 = 68,
+
+    /// The `eth` protocol version 69.
+    Eth69 = 69,
 }
 
 impl EthVersion {
     /// The latest known eth version
-    pub const LATEST: Self = Self::Eth68;
+    pub const LATEST: Self = Self::Eth69;
```
[the remark this answers is not in the document]
Comment (malik672) [reply crates/net/eth-wire-types/src/version.rs:34]: @mattsse yes, I know. I'm writing this one for eventually when eth69 gets introduced, will complete the new types in a separate branch and pull a pr now, can I make this a draft till then?

**Comment by onbjerg** [comment]
Hey @malik672, what's the current status on this? :blush:

**Comment by emhane** [comment]
could https://github.com/paradigmxyz/reth/issues/10659 pls be addressed in many small prs? think it shouldn't be a problem to get the code compiling without doing all at once

**Review by mattsse — COMMENTED** [review]
the 69 is still in the works and is breaking, so we can't really apply it like this.

because this drifted quite a bit, I'm closing this because I think it's easier to first introduce the new 69 types before integrating it

**Pull request closed without merging**